### ISSUE TITLE
Error building ios archive in xcode

### ISSUE THREAD

**Issue by bartwaggoner** [flutter/flutter#13364]
I am following the flutter docs on releasing an app for ios precisely, but get the following error when generating the archive in xcode (it can't find pod libs for reachability etc).  After spinning my wheels in the mud for half a day, can anyone point me in the direction of resolving this?

```
ld: warning: directory not found for option '-L/Users/bartwaggoner/Library/Developer/Xcode/DerivedData/Runner-duuffrseacvinacmifrehqvwbbxo/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/Reachability'
ld: warning: directory not found for option '-L/Users/bartwaggoner/Library/Developer/Xcode/DerivedData/Runner-duuffrseacvinacmifrehqvwbbxo/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/connectivity'
ld: warning: directory not found for option '-L/Users/bartwaggoner/Library/Developer/Xcode/DerivedData/Runner-duuffrseacvinacmifrehqvwbbxo/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/path_provider'
ld: warning: directory not found for option '-L/Users/bartwaggoner/Library/Developer/Xcode/DerivedData/Runner-duuffrseacvinacmifrehqvwbbxo/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/url_launcher'
ld: library not found for -lReachability
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

(note flutter, xcode up to date as of today and flutter doctor shows no probs - i can run in simulator fine)

**Comment by eseidelGoogle** [comment]
@xster or @cbracken would know.

**Comment by bartwaggoner** [comment]
I notice in my xcode project that the "Pods" folder is empty, unllike the "Flutter", "Runner" etc. folders. Is this the problem? In my Intellij project the ios/Pods folder is not empty and has Reachability etc. in it. If this is not getting exporte to the xcode project how do I change this?

**Comment by bartwaggoner** [comment]
OK, I got farther by loading the .xcworkspace instead of .xccodeproj (following the instructions 'precisely' helps ;-) Now I get a series of linker errors. Here is one such:

Undefined symbols for architecture armv7:
  "_OBJC_CLASS_$_FlutterError", referenced from:
      objc-class-ref in liburl_launcher.a(UrlLauncherPlugin.o)
  "_FlutterMethodNotImplemented", referenced from:
      -[ConnectivityPlugin handleMethodCall:result:] in libconnectivity.a(ConnectivityPlugin.o)
      ___47+[FLTPathProviderPlugin registerWithRegistrar:]_block_invoke in libpath_provider.a(PathProviderPlugin.o)
      ___43+[UrlLauncherPlugin registerWithRegistrar:]_block_invoke in liburl_launcher.a(UrlLauncherPlugin.o)
  "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in libconnectivity.a(ConnectivityPlugin.o)
      objc-class-ref in libpath_provider.a(PathProviderPlugin.o)
      objc-class-ref in liburl_launcher.a(UrlLauncherPlugin.o)
  "_OBJC_METACLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_METACLASS_$_AppDelegate in AppDelegate.o
  "_OBJC_CLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_CLASS_$_AppDelegate in AppDelegate.o
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by bartwaggoner** [comment]
Here is the complete list of link errors - any help for this would be greatly appreciated!:

Showing All Errors Only
  "_OBJC_CLASS_$_FlutterError", referenced from:
      objc-class-ref in liburl_launcher.a(UrlLauncherPlugin.o)
  "_FlutterMethodNotImplemented", referenced from:
      -[ConnectivityPlugin handleMethodCall:result:] in libconnectivity.a(ConnectivityPlugin.o)
      ___47+[FLTPathProviderPlugin registerWithRegistrar:]_block_invoke in libpath_provider.a(PathProviderPlugin.o)
      ___43+[UrlLauncherPlugin registerWithRegistrar:]_block_invoke in liburl_launcher.a(UrlLauncherPlugin.o)
  "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in libconnectivity.a(ConnectivityPlugin.o)
      objc-class-ref in libpath_provider.a(PathProviderPlugin.o)
      objc-class-ref in liburl_launcher.a(UrlLauncherPlugin.o)
  "_OBJC_METACLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_METACLASS_$_AppDelegate in AppDelegate.o
  "_OBJC_CLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_CLASS_$_AppDelegate in AppDelegate.o
ld: symbol(s) not found for architecture armv7

**Comment by bartwaggoner** [comment]
OK! Got it to build by choosing Deployment Target=11.2

How can I build something that works on iOS 9 or 10 and later?

**Comment by bartwaggoner** [comment]
OK, another update! I got the archive to build with ios target 9.0 and later by setting "Valid architectures" to only "arm64" (I removed armv7 armv7s).  Is this an OK solution?

**Comment by cbracken** [comment]
@bartwaggoner apologies for the delay over the break. Yep that is the correct solution!

Closing but if you're still blocked on this, let us know!

**Comment by acidjazz** [comment]
I'm getting this error as well, which was the right solution? he gave two. both haven't worked for me.

**Comment by xster** [comment]
@acidjazz can you create a new bug with your details and `flutter doctor` output?

**Comment by acidjazz** [comment]
@xster well its pretty similar, i also have a folder not created for a package I'm using

```
ld: warning: directory not found for option '-L/Users/k/Library/Developer/Xcode/DerivedData/Runner-euwhfipnbjdjrxaeukvbwksvhgrt/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/image_picker'
ld: warning: directory not found for option '-L/Users/k/Library/Developer/Xcode/DerivedData/Runner-euwhfipnbjdjrxaeukvbwksvhgrt/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/shared_preferences'
ld: library not found for -limage_picker
```

**Comment by acidjazz** [comment]
@xster while creating a new issue and documenting the steps to do so my error has disappeared, because ofcourse :)

**Comment by Brijeshlakkad** [comment]
I have problem in linking with python files in starcore_for_ios directory.
Please see this [problem](https://github.com/srplab/starcore_for_flutter/issues/4#issuecomment-445547245)

**Comment by raghaaavvvv** [comment]
I figured out the problem, all of the deployment targets need to be the same. That is, in the project file, the target, and the pods project as well.

**Comment by lockelost** [comment]
@iamatulkumar Can you tell me more about it, please? 
My case is the compilation fails, saying the plugin header files are not found.
Build for the physical device or Generic devie works fine.
